client/tokenized-payment-request/payment-request.js

```diff
@@ -0,0 +1,406 @@
+	displayLoginConfirmationDialog,
+	getPaymentRequestData,
+	waitForAction,
+} from './frontend-utils';
+import PaymentRequestCartApi from './cart-api';
+import debounce from './debounce';
+
+const noop = () => null;
+
+/**
+ * Class to handle Stripe payment forms.
+ */
+export default class WcpayPaymentRequest {
+	/**
+	 * Whether the payment was aborted by the customer.
+	 */
+	paymentAborted = false;
+
+	/**
+	 * Whether global listeners have been added.
+	 */
+	areListenersInitialized = false;
+
+	/**
+	 * The cart data represented if the product were to be added to the cart (or, on cart/checkout pages, the cart data itself).
+	 * This is useful on product pages to understand if shipping is needed.
+	 */
+	cachedCartData = undefined;
+
+	/**
+	 * API to interface with the cart.
+	 *
+	 * @type {PaymentRequestCartApi}
+	 */
+	paymentRequestCartApi = undefined;
+
+	/**
+	 * WCPayAPI instance.
+	 *
+	 * @type {WCPayAPI}
+	 */
+	wcpayApi = undefined;
+
+	/**
+	 * On page load for product pages, we might get some data from the backend (which might get overwritten later).
+	 */
+	initialProductData = undefined;
+
+	constructor( { wcpayApi, paymentRequestCartApi, productData } ) {
+		this.wcpayApi = wcpayApi;
+		this.paymentRequestCartApi = paymentRequestCartApi;
+		this.initialProductData = productData;
+	}
+
+	/**
+	 * Starts the payment request
+	 */
+	async startPaymentRequest() {
+		// reference to this class' instance, to be used inside callbacks to avoid `this` misunderstandings.
+		const _self = this;
+		// TODO: is this creating multiple handlers to events on different `paymentRequest` objects?
+		const paymentRequest = getPaymentRequest( {
+			stripe: this.wcpayApi.getStripe(),
+			cartData: this.cachedCartData,
+			productData: this.initialProductData,
+		} );
+
+		// Check the availability of the Payment Request API first.
+		const paymentPermissionResult = await paymentRequest.canMakePayment();
+		if ( ! paymentPermissionResult ) {
+			doAction( 'wcpay.payment-request.availability', {
+				paymentRequestType: null,
+			} );
+			return;
+		}
+
+		// TODO: Don't display custom button when paymentRequestType is `apple_pay` or `google_pay`.
+		let buttonBranding = null;
+		if ( paymentPermissionResult.applePay ) {
+			buttonBranding = 'apple_pay';
+		} else if ( paymentPermissionResult.googlePay ) {
+			buttonBranding = 'google_pay';
+		} else {
+			buttonBranding = 'payment_request_api';
+		}
+
+		doAction( 'wcpay.payment-request.availability', {
+			paymentRequestType: buttonBranding,
+		} );
+
+		setPaymentRequestBranding( buttonBranding );
+		trackPaymentRequestButtonLoad(
+			wcpayPaymentRequestParams.button_context
+		);
+
+		// On PDP pages, we need to use an anonymous cart to check out.
+		// On cart, checkout, place order pages we instead use the cart itself.
+		if ( wcpayPaymentRequestParams.button_context === 'product' ) {
+			this.paymentRequestCartApi.createAnonymousCart().then( noop );
+		}
+
+		const prButton = this.wcpayApi
```

Comment:
```suggestion
		const paymentRequestButton = this.wcpayApi
```
[Just sayin'](https://developer.wordpress.org/coding-standards/wordpress-coding-standards/javascript/#abbreviations-and-acronyms). 🤷

Reply:
Just another one of those things I had copied over without giving it a second look 😄 654097724